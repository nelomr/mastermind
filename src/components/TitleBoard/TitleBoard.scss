.title-board {
  &.container {
    margin-bottom: 40px;
  }
}