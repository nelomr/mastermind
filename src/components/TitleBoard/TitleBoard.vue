<template>
  <div class="title-board container">
    <h1>Mastermind</h1>
    <div v-if="game" v-text="`#${game.reference}`"></div>
  </div>
</template>
<script>
import { mapState } from 'vuex';

export default {
  name: 'Title',
  computed: {
    ...mapState([
      'game'
    ])
  },
}
</script>
<style lang="scss">
  @import "./TitleBoard.scss";
</style>