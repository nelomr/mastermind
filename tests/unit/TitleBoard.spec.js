import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import TitleBoard from '@/components/TitleBoard/TitleBoard.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('TitleBoard.vue', () => {
  let game = null;

  let store;
  let state;

  beforeEach(() => {
   
    state = {
      game
    };
    store = new Vuex.Store({
      state
    });
  });
  const build = () => {
    const wrapper = shallowMount(TitleBoard, {
      mocks: {$store: store}
    });

    return {
        wrapper
    };
  };

  it('renders the component list', () => {
    const {wrapper} = build();

    expect(wrapper.html()).toMatchSnapshot();
  });
})
