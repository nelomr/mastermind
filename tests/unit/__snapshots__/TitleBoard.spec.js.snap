// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBoard.vue renders the component list 1`] = `
<div class="title-board container">
  <h1>Mastermind</h1>
  <!---->
</div>
`;
